sentry-android-timber/src/main/java/io/sentry/android/timber/SentryTimberTree.kt

```diff
@@ -17,42 +17,218 @@ class SentryTimberTree(
+    override fun w(t: Throwable?) {
+        logWithSentry(Log.WARN, t, null)
+    }
+
+    /** Log an error message with optional format args. */
+    override fun e(
+        message: String?,
+        vararg args: Any?
+    ) {
+        logWithSentry(Log.ERROR, null, message, *args)
+    }
+
+    /** Log an error exception and a message with optional format args. */
+    override fun e(
+        t: Throwable?,
+        message: String?,
+        vararg args: Any?
+    ) {
+        logWithSentry(Log.ERROR, t, message, *args)
+    }
+
+    /** Log an error exception. */
+    override fun e(t: Throwable?) {
+        logWithSentry(Log.ERROR, t, null)
+    }
+
+    /** Log an assert message with optional format args. */
+    override fun wtf(
+        message: String?,
+        vararg args: Any?
+    ) {
+        logWithSentry(Log.ASSERT, null, message, *args)
+    }
+
+    /** Log an assert exception and a message with optional format args. */
+    override fun wtf(
+        t: Throwable?,
+        message: String?,
+        vararg args: Any?
+    ) {
+        logWithSentry(Log.ASSERT, t, message, *args)
+    }
+
+    /** Log an assert exception. */
+    override fun wtf(t: Throwable?) {
+        logWithSentry(Log.ASSERT, t, null)
+    }
+
+    /** Log at `priority` a message with optional format args. */
+    override fun log(
+        priority: Int,
+        message: String?,
+        vararg args: Any?
+    ) {
+        logWithSentry(priority, null, message, *args)
+    }
+
+    /** Log at `priority` an exception and a message with optional format args. */
+    override fun log(
+        priority: Int,
+        t: Throwable?,
+        message: String?,
+        vararg args: Any?
+    ) {
+        logWithSentry(priority, t, message, *args)
+    }
+
+    /** Log at `priority` an exception. */
+    override fun log(
+        priority: Int,
+        t: Throwable?
+    ) {
+        logWithSentry(priority, t, null)
+    }
+
+    override fun log(
+        priority: Int,
+        tag: String?,
+        message: String,
+        t: Throwable?
+    ) {
+        // no-op as we've overridden all the methods
+    }
+
+    private fun logWithSentry(
+        priority: Int,
+        throwable: Throwable?,
+        message: String?,
+        vararg args: Any?
+    ) {
+        if (message.isNullOrEmpty() && throwable == null) {
+            return // Swallow message if it's null and there's no throwable
+        }
 
-    /**
-     * Captures a Sentry Event if the min. level is equal or higher than the min. required level.
-     */
-    override fun log(priority: Int, tag: String?, message: String, throwable: Throwable?) {
         val level = getSentryLevel(priority)
+        val sentryMessage = Message().apply {
+            this.message = message
+            if (message != null && args.isNotEmpty()) {
+                this.formatted = message.format(*args)
+            }
+            this.params = args.mapNotNull { it.toString() }
```

Comment:
nit: what about we map `null` to `"null"`? Could "collide" with an actual string that was "null" but the goal is to indicate what values where provided vs what wasn't. (%s, %s, %s = null, "bla", null), args would be : ["bla"] so not sure what position had the value vs null?

Alternatively we can omit args, not sure how much difference is in the UI at the end though

Reply:
Makes total sense. Actually `mapNotNull` is useless here anyway, because, at first, it maps, and then appends the values that are not-null, but in our case we transform everything `toString()` so there will never be a null value. Will change it to `map` 👍 

I've been following the logback implementation though, and there it's filter not-null values, not sure if should be also changed? https://github.com/getsentry/sentry-java/blob/5d8208dba4a7eb32696e38047ab21a4c7d435f3d/sentry-logback/src/main/java/io/sentry/logback/SentryAppender.java#L100